File: backend/app/models/video.py
from typing import Literal

from pydantic import BaseModel, Field, HttpUrl


class Video(BaseModel):
    """A video record supplied by the local video source file."""

    id: str
    title: str
    description: str
    url: HttpUrl
    thumbnail_url: HttpUrl | Literal[""]
    likes: int = Field(ge=0)
    shares: int = Field(ge=0)


class VideosResponse(BaseModel):
    """Response payload for the video carousel."""

    videos: list[Video]

File: frontend/frontend/types/video.ts
export type Video = {
  id: string
  title: string
  description: string
  url: string
  thumbnailUrl: string
  creator: string
  role: string
  likes: number
  shares: number
  duration: string
  accent: string
  posterLabel: string
}

export type ApiVideo = {
  id: string
  title: string
  description: string
  url: string
  thumbnail_url: string
  likes: number
  shares: number
}

export type VideosResponse = {
  videos: ApiVideo[]
}
